x-pack/plugins/cloud_security_posture/server/saved_objects/migrations/csp_rule_template.ts

```diff
@@ -36,6 +37,20 @@ function migrateCspRuleMetadata(
   };
 }
 
+function deprecateCspRuleTemplateState(
```

Comment:
So I really thought about this point when I prepare the PR.

I agree that naming convention has huge importance in code, and the reason I named the function that way and not by using `migration` is that the context is migration, meaning the function is located in the migration directory.

But anyway, we obviously can choose a better name.

I changed to `migrateCspRuleTemplateDeprecateState`.

WDYT?

Reply:
my 2 cents: 

- we have 2 migrations currently, both with names to match what they do. 
- at some point, i guess, we're going to have to migrate something that explaining it in the function name would be kinda annoying. 

to avoid that and keep things consistent, my suggestion is to rename all the migrations functions to names that match their version, like so:
```diff
+ export const cspRuleTemplateMigrations: SavedObjectMigrationMap = {
+  '8.4.0': migrate840,
+ };
```

Reply:
@orouz Interesting point!

I agree that naming the migration can be annoying, but on the other hand, if the name is accurate and self-explanatory, it can save time for the reader.

But it is a legitimate point for sure! plus there are many other plugins that use what Or suggests.

@kfirpeled, @JordanSh, @CohenIdo  what do you think?

Reply:
go with Or's naming suggestion and add a comment to helping the reader